addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/InformationDisclosureReferrerScanRule.java

```diff
@@ -125,33 +125,32 @@ private boolean isRequestedURLSameDomainAsHTTPReferrer(String host, String refer
     }
 
     private AlertBuilder buildAlert(String evidence, String other) {
-        return newAlert()
-                .setRisk(getRisk())
+        return baseAlert()
                 .setConfidence(Alert.CONFIDENCE_MEDIUM)
-                .setDescription(getDescription())
                 .setOtherInfo(other)
-                .setSolution(getSolution())
-                .setEvidence(evidence)
-                .setCweId(getCweId())
-                .setWascId(getWascId());
+                .setEvidence(evidence);
     }
 
     private AlertBuilder buildCcAlert(String evidence, String other, BinRecord binRec) {
         if (binRec != null) {
             other = other + '\n' + getBinRecString(binRec);
         }
-        return newAlert()
-                .setRisk(getRisk())
+        return baseAlert()
                 .setConfidence(binRec != null ? Alert.CONFIDENCE_HIGH : Alert.CONFIDENCE_MEDIUM)
-                .setDescription(getDescription())
                 .setOtherInfo(other)
-                .setSolution(getSolution())
-                .setEvidence(evidence)
-                .setCweId(getCweId())
-                .setWascId(getWascId());
+                .setEvidence(evidence);
+    }
+
+    private AlertBuilder baseAlert() {
```

Comment:
This could be done in `buildAlert` instead, the `buildCcAlert` can pass in the evidence/other and override the confidence after.